src/earn/hooks.ts

```diff
@@ -94,3 +90,52 @@ export function useDepositEntrypointInfo({
   }, [asyncExchanges.result])
   return { hasDepositToken, hasTokensOnSameNetwork, hasTokensOnOtherNetworks, canCashIn, exchanges }
 }
+
+export function usePrepareTransactions(
+  mode: EarnActiveMode,
+  additionalArgs?: { rewardsPositions: Position[] }
+) {
+  const getTransactionFunction = () => {
+    switch (mode) {
+      case 'deposit':
+      case 'swap-deposit':
+        return prepareDepositTransactions
+      case 'withdraw':
+        return prepareWithdrawTransactions
+      case 'claim-rewards':
+        return prepareClaimTransactions
+      case 'exit':
+        return prepareWithdrawAndClaimTransactions
```

Comment:
having all the modes here is nice, but I wonder if it does more harm that good. This hook I believe is mainly for enter amount screen which needs the functionality to refresh transactions as the user input changes, but for claim and exit do we even need it (related to my other comment)? Wonder if this would be simpler to just wrap the transaction preparation in a `useAsync` like how it's done currently. You also wouldn't need the args hack you have below.